### ISSUE TITLE
Plugin Dependencies Support

### ISSUE THREAD

**Issue by Luc45** [wp-cli/extension-command#407]
## Feature Request

WordPress 6.5 adds the [Plugin Dependencies](https://make.wordpress.org/core/2024/02/15/merge-announcement-plugin-dependencies/) capability. It would be useful if `plugin install` had a `--with-dependencies` flag to also install the dependencies of the plugin being installed.

**Comment by swissspidy** [comment]
Generally speaking, we need to test with the new plugin dependencies feature and see if and what to change. If the plugin‘s API provides this information easily, then I could see such a flag being useful for sure.

**Comment by Luc45** [comment]
I haven't looked at how WP Core does it, but it should probably be surfaced on the API already to render it on the wp-admin?plugins.php page

**Comment by swissspidy** [comment]
Right. This is surfaced in the API via the `requires_plugins` field which contains an array of slugs.

Example: https://api.wordpress.org/plugins/info/1.2/?action=plugin_information&request%5Blocale%5D=en_US&request%5Bslug%5D=wpsso-wc-metadata&fields=versions

So should be available for us too 👍

**Comment by swissspidy** [comment]
Listing dependencies in `wp plugin get <slug>` would also be useful
